src/data_access/JSONPersistence.java

```diff
@@ -0,0 +1,155 @@
+            /* Add the tag as the key and the array as the values. */
+            obj.put(tag, array);
+        }
+        return obj;
+    }
+
+    /**
+     * Create a JSONArray form the given User object's favourites.
+     * @param user the User object which we work with.
+     * @return JSONArray that contains a list of all the IDs for the recipes
+     * that the user has "favourited".
+     */
+    private JSONArray JSONForUserFavourites(User user) {
+        JSONArray array = new JSONArray();
+        ArrayList<Recipe> recipeList = user.getFavourites();
+        for (Recipe recipe : recipeList) {
+            array.put(recipe.getMealId());
+        }
+        return array;
+    }
+
+    /**
+     * Create a JSONObject that contains the user's tags (with associated
+     * recipes) and favourites.
+     * @param user The User object on which we base the output
+     * @return A JSONObject that has two keys: "favourites" and "tags".
+     * The value for "favourites" is a JSONArray of the IDs of the
+     * "favourited" recipes. The value for "tags" is a JSONObject, which
+     * contains the tag names as keys, and JSONArrays of the IDs of the
+     * recipes as the values.
+     */
+    private JSONObject constructJSONObjectFromUser(User user) {
+        JSONObject obj = new JSONObject();
+        JSONObject tagsJSON = JSONForUserTags(user);
+        JSONArray favouritesJSON = JSONForUserFavourites(user);
+        obj.put("username", user.getUsername());
+        obj.put("tags", tagsJSON);
+        obj.put("favourites", favouritesJSON);
+        return obj;
+    }
+
+    /**
+     * Save the user's tags and favourites to the file that was given upon
+     * object creation.
+     * @param user the User object whose contents are saved.
+     */
+    public void save(User user) {
+        JSONObject obj = constructJSONObjectFromUser(user);
+        BufferedWriter writer;
+        try {
+            writer = new BufferedWriter(new FileWriter(this.filePath));
+            writer.write(obj.toString(4));
+            writer.close();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public User load() {
+        User user;
+        String name;
+        JSONArray favourites;
+        JSONObject tagsMap;
+        /* Read and parse the information from the file. */
+        try {
+            String fileContents = Files.readString(Path.of(this.filePath));
+            JSONTokener tokenizer = new JSONTokener(fileContents);
+            JSONObject obj = new JSONObject(tokenizer);
+            name = obj.getString("username");
+            favourites = obj.getJSONArray("favourites");
+            tagsMap = obj.getJSONObject("tags");
+        } catch (JSONException | IOException e) {
+            throw new RuntimeException(e);
+        }
+        /* Construct the user object. */
+        /* TODO: fix this hack, where we don't use the password. */
+        user = userFactory.create(name, "");
```

Comment:
As a note, the change to the User Factory file to remove the password parameter from its create method has been taken care of and pushed to the main branch.